content/04.workflows-and-software.md

```diff
@@ -80,36 +79,56 @@ Analysis workflows commonly rely on multiple software packages to generate final
-
-Software management systems perform some combination of software installation, management, and packaging that alleviate problems that arise from dependencies and facilitate documentation of software versions.
 On many compute systems, system-wide software management is overseen by system administrators, who ensure commonly-used and requested software is installed into a "module" system available to all users.
-Unfortunately, this system does not lend itself well for exploring new workflows and software, as researchers do not have permission to install software themselves.
-The Conda package manager has emerged as a leading solution, largely because it handles both cluster permission and version conflict issues with a user-based software environment system, and features a straightforward "recipe" system which simplifies the process of making new software installable (**Figure @fig:conda_figure_ABC**).
-Conda enables lightweight software installation and can be used with the same commands across platforms, but can still be impacted by differences in the host operating system.
-Alternatively, wrapping software environments in "containers" that capture and reproduce all other aspects of the runtime environment can enhance reproducibility over time [@doi:10.1016/j.cels.2018.03.014].
-Container-based software installation via docker and singularity is common for production-level workflows.
+Unfortunately, this system limits software version transparency and does not lend itself well to exploring new workflows and software, as researchers do not have permission to install software themselves.
+To meet this need, most workflow managers integrate with software management systems that handle software installation, management, and packaging, alleviating problems that arise from complex dependencies and facilitating documentation of software versions.
+Software management systems range from lightweight systems that manage only the software and its dependencies, to heavyweight systems that control for all aspects of the runtime and operating system, ensuring 100% reproducibility of results across computational platforms and time.
+
+On the lightweight end, the conda package manager has emerged as a leading software management solution for research workflows (**Figure @fig:conda_figure_ABC**).
+Conda handles both cluster permission and version conflict issues with a user-based software environment system, and features a straightforward "recipe" system which simplifies the process of making new software installable (including simple management of versions and updates).
+These features have led to widespread adoption within the bioinformatic community: packages for new software become quickly available, and can be installed easily across platforms.
+However, conda does not attempt to isolate software installations or standardize the runtime environment, meaning installations will not be completely reproducible over time.
```

Comment:
hmm, this:
>However, conda does not attempt to isolate software installations or standardize the runtime environment, meaning installations will not be completely reproducible over time.

doesn't quite make sense to me. Conda _does_ isolate software installs, just not perfectly. And I don't know what "standardize the runtime environment" means, although I can guess.

Maybe:
> However, conda does not completely isolate software installations and aims neither for bitwise reproducibility nor long-term archiving of install packages, meaning installations will not be completely reproducible over time.

Reply:
yes, much better -thanks!